packages/lab/src/number-input/NumberInput.tsx

```diff
@@ -418,10 +481,52 @@ export const NumberInput = forwardRef<HTMLDivElement, NumberInputProps>(
       inputOnKeyDown?.(event);
     };
 
-    const handleBeforeInput = () => {
-      setIsEditing(true);
+    const handleIncrementMouseDown = (
+      event: SyntheticEvent,
+      disableIncrement: boolean,
+    ) => {
+      event.preventDefault();
+      if (!disableIncrement) {
+        setIsEditing(false);
+        activateIncrement(event);
+      } else if (inputRef.current) {
+        inputRef.current.select();
+      }
+    };
+
+    const handleDecrementMouseDown = (
+      event: SyntheticEvent,
+      disableDecrement: boolean,
+    ) => {
+      event.preventDefault();
+      if (!disableDecrement) {
+        setIsEditing(false);
+        activateDecrement(event);
+      } else if (inputRef.current) {
+        inputRef.current.select();
+      }
+    };
+
+    const handleContainerMouseUp: MouseEventHandler<HTMLDivElement> = (
+      event,
+    ) => {
+      setIsFocused(true);
+      restProps.onMouseUp?.(event);
```

Comment:
This is not enough, we still need to destructure onMouseUp when it's passed in as it still will be overwritten.